pkg/ccl/backupccl/show.go

```diff
@@ -829,14 +813,24 @@ func backupShowerFileSetup(inCol tree.Expr) backupShower {
 		{Name: "end_key", Typ: types.Bytes},
 		{Name: "size_bytes", Typ: types.Int},
 		{Name: "rows", Typ: types.Int},
+		{Name: "locality", Typ: types.String},
 	},
 
-		fn: func(manifests []BackupManifest, manifestDirs []string) (rows []tree.Datums, err error) {
-			if (inCol != nil) && len(manifestDirs) == 0 {
-				return nil, errors.AssertionFailedf(
-					"manifestDirs empty even though backup is in collection")
+		fn: func(info backupInfo) (rows []tree.Datums, err error) {
+
+			var manifestDirs []string
+			var localityAware bool
+			if len(inCol) > 0 {
+				manifestDirs, err = getManifestDirs(info.subdir, info.defaultURIs)
+				if err != nil {
+					return nil, err
+				}
+
+				if len(info.localityInfo[0].URIsByOriginalLocalityKV) > 0 {
```

Comment:
just a sanity check, we expect info.localityInfo to always have atleast one element?

Reply:
yes, the length is equal to the number of backups in the chain.